server/src/main/java/org/elasticsearch/common/xcontent/XContentBuilder.java

```diff
@@ -110,13 +112,28 @@ public static XContentBuilder builder(XContent xContent, Set<String> includes, S
         writers.put(Text.class, (b, v) -> b.value((Text) v));
 
         WRITERS = Collections.unmodifiableMap(writers);
+
+        Map<Class<?>, HumanReadableTransformer> humanReadableTransformer = new HashMap<>();
+        // These will be moved to a different class at a later time to decouple them from XContentBuilder
+        humanReadableTransformer.put(TimeValue.class, v -> ((TimeValue) v).millis());
+        humanReadableTransformer.put(ByteSizeValue.class, v -> ((ByteSizeValue) v).getBytes());
+
+        HUMAN_READABLE_TRANSFORMERS = Collections.unmodifiableMap(humanReadableTransformer);
     }
 
     @FunctionalInterface
-    private interface Writer {
+    interface Writer {
         void write(XContentBuilder builder, Object value) throws IOException;
     }
 
+    /**
+     * Interface for transforming complex objects into their "raw" equivalents for human-readable fields
+     */
+    @FunctionalInterface
+    interface HumanReadableTransformer {
```

Comment:
Do you think this should be an interface the thing implements rather than a transformer to look up? That feels a little more explicit.

Reply:
With an interface we can't extend classes we don't control (Lucene or Joda ones)

Reply:
But we could add both if we wanted? I can do that in subsequent work if you think it'd be worth adding

Reply:
You are right. I don't think it is worth adding multiple ways to do this.